bluemira/equilibria/fem_fixed_boundary/equilibrium.py

```diff
@@ -0,0 +1,266 @@
+            "kappa_95", kappa_95, "dimensionless", source
+        )
+        plasmod_options["params"].set_parameter(
+            "delta_95", delta_95, "dimensionless", source
+        )
+
+        plasmod_solver = PlasmodTransportSolver(
+            params=plasmod_options["params"],
+            build_config=plasmod_options["build_config"],
+        )
+        plasmod_solver.execute()
+
+        if plot:
+            plot_profile(
+                plasmod_solver.x, plasmod_solver.pprime(plasmod_solver.x), "pprime", "-"
+            )
+            plot_profile(
+                plasmod_solver.x, plasmod_solver.ffprime(plasmod_solver.x), "ffrime", "-"
+            )
+
+        # generate mesh for the Grad-Shafranov solver
+        plasma.shape.boundary[0].mesh_options = {
+            "lcar": lcar_coarse,
+            "physical_group": "lcfs",
+        }
+        plasma.shape.mesh_options = {
+            "lcar": lcar_coarse,
+            "physical_group": "plasma_face",
+        }
+
+        m = meshing.Mesh()
+        m(plasma)
+
+        msh_to_xdmf("Mesh.msh", dimensions=(0, 2), directory=".", verbose=True)
+
+        mesh, boundaries, subdomains, labels = import_mesh(
+            "Mesh",
+            directory=".",
+            subdomains=True,
+        )
+
+        # initialize the Grad-Shafranov solver
+        p_order = gs_options["p_order"]
+        gs_solver = FemGradShafranovFixedBoundary(mesh, p_order=p_order)
+
+        bluemira_print("Solving fixed boundary Grad-Shafranov...")
+
+        gs_solver.solve(
+            plasmod_solver.pprime,
+            plasmod_solver.ffprime,
+            plasmod_solver.I_p,
+            tol=gs_options["tol"],
+            max_iter=gs_options["max_iter"],
+            relaxation=gs_relaxation,
+            verbose_plot=gs_options["verbose_plot"],
+        )
+
+        # create the finer mesh to calculate the isofluxes
+        plasma.shape.boundary[0].mesh_options = {
+            "lcar": lcar_fine,
+            "physical_group": "lcfs",
+        }
+        plasma.shape.mesh_options = {"lcar": lcar_fine, "physical_group": "plasma_face"}
+
+        m = meshing.Mesh()
+        m(plasma)
+
+        msh_to_xdmf("Mesh.msh", dimensions=(0, 2), directory=".", verbose=verbose)
+
+        mesh, boundaries, subdomains, labels = import_mesh(
+            "Mesh",
+            directory=".",
+            subdomains=True,
+        )
+
+        points = mesh.coordinates()
+        x2d_data = np.array([gs_solver.psi_norm_2d(x) for x in points])
+
+        # calculate kappa_95 and delta_95
+        r_geo, kappa_95, delta_95 = calculate_plasma_shape_params(
+            points, x2d_data, [np.sqrt(0.95)]
+        )
+        r_geo, kappa_95, delta_95 = r_geo[0], kappa_95[0], delta_95[0]
+
+        # calculate the iteration error
+        err_delta = abs(delta_95 - delta95_t) / delta95_t
+        err_kappa = abs(kappa_95 - kappa95_t) / kappa95_t
+        iter_err = max(err_delta, err_kappa)
```

Comment:
Perhaps something like `np.hypot(err_delta, err_kappa)` would be more stable here.

Reply:
It doesn't seem to make much different, but worth keeping in mind when trying to find a stable scheme